client/analytics/report/downloads/config.js

```diff
@@ -126,10 +119,13 @@ export const advancedFilters = {
 			input: {
 				component: 'Search',
 				type: 'orders',
-				getLabels: getRequestByIdString( NAMESPACE + 'orders', order => ( {
-					id: order.id,
-					label: '#' + order.id,
-				} ) ),
+				getLabels: async value => {
+					const orderIds = value.split( ',' );
+					return await orderIds.map( orderId => ( {
+						id: orderId,
+						label: '#' + orderId,
+					} ) );
+				},
```

Comment:
This change is not related to the others, but I saw we could avoid an API request here.